Mathlib/Combinatorics/SimpleGraph/Path.lean

```diff
@@ -1072,6 +1072,22 @@ theorem mem_supp_iff (C : G.ConnectedComponent) (v : V) :
     v ∈ C.supp ↔ G.connectedComponentMk v = C :=
   Iff.rfl
 
+lemma mem_supp_iff_of_adj {v w : V} (c : G.ConnectedComponent) (hadj : G.Adj v w) :
```

Comment:
Another option:
```suggestion
lemma mem_supp_congr_adj {v w : V} (c : G.ConnectedComponent) (hadj : G.Adj v w) :
```
Not sure which one I like best

Reply:
I think this one is more elegant, so I'll adopt it:)